File: Java/src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author jorge
 */
public class Monster extends LabyrinthCharacter{
    private static final int INITIAL_HEALTH = 5;
    
    public Monster(String nombre, float i, float s){
        super(nombre,i,s,INITIAL_HEALTH);
    }
       
    @Override
    public float attack(){
        return Dice.intensity(getStrength());
    }
    
    @Override
    public String toString(){
        return "Monster"+super.toString();
    }
    
    @Override
    public boolean defend(float receivedAttack){
        boolean isDead = dead();
        
        if (!isDead){
            float defensiveEnergy = Dice.intensity(getIntelligence());
            if (defensiveEnergy < receivedAttack){
                gotWounded();
                isDead=dead();
            }
        }
        
        return isDead;
    }
}

File: Ruby/Monster.rb
#encoding: UTF-8

require_relative 'Dice'
require_relative 'Weapon'
require_relative 'Shield'
require_relative 'LabyrinthCharacter'

module Irrgarten
	class Monster < LabyrinthCharacter
		@@INITIAL_HEALTH=5
		
		public	
		def initialize(n,i,s)
			super(n,i,s,@@INITIAL_HEALTH)
		end
		
		def attack()
			Dice.intensity(@strength)
		end
		
		def to_string()
			"Monster"+super
		end
		
		def defend(received_attack)
			is_dead=self.dead()
			if (!is_dead)
				defensive_energy = Dice.intensity(@intelligence)
				if (defensive_energy < received_attack)
					self.got_wounded()
					is_dead=self.dead()
				end
			end
			
			is_dead
		end
		
		public_class_method :new
	end
end
